File: python/Leetcode589.py
from typing import List
from Utils import Node


class Leetcode589:
    def preorder(self, root) -> List[int]:
        visited = list()       
        if root is None:
            return [] 
        self.traverse(root, visited)
        return visited

    def traverse(self, root, visited):
        visited.append(root.val)
        for child in root.children:
            self.traverse(child, visited)

File: java/src/Leetcode589.java
import utils.Node;
import java.util.*;

class Leetcode589 {
    public List<Integer> preorder(Node root){
        List<Integer> visited = new ArrayList<>();
        traverse(root, visited);
        return visited;
    }

    private void traverse(Node root, List<Integer> visited){
        visited.add(root.val);
        for(Node node : root.children){
            traverse(node, visited);
        }
    }
}